### PR TITLE
refactor(network): updates to support buf lint and generation in opin…

### PR DESCRIPTION
**Pull request by sandersms**
…etcommon.

This switches the generation of the protobuf files for go and python to use "buf generate" instead of standard protoc.  The makefile is updated to support buf lint, buf generate, and document generation.

### CODE REVIEW

**Comment by sandersms** [comment]
Note that buf generate can't be run in a docker container due to protoc not being included.

**Comment by artek-koltun** [comment]
> Note that buf generate can't be run in a docker container due to protoc not being included.

Using the Docker image offers numerous advantages (consistent build env, pinned tool version etc.). In case it's not feasible to avoid utilizing protoc in the build, have you considered the possibility of extending the buf image with protoc? It seems that we may just need to include a single package to that image, as discussed in [this GitHub issue](https://github.com/bufbuild/buf/issues/1037).

**Comment by sandersms** [comment]
> > Note that buf generate can't be run in a docker container due to protoc not being included.
> 
> Using the Docker image offers numerous advantages (consistent build env, pinned tool version etc.). In case it's not feasible to avoid utilizing protoc in the build, have you considered the possibility of extending the buf image with protoc? It seems that we may just need to include a single package to that image, as discussed in [this GitHub issue](https://github.com/bufbuild/buf/issues/1037).

Thanks for the different issue link - I had found some, but not one where it indicated the combined build.  I will incorporate that for the generate operation as was our desired outcome with using buf.

File: network/opinetcommon/Makefile [hunk 1]
```diff
@@ -2,22 +2,26 @@
 # Copyright (C) 2022 Intel Corporation
 # Copyright (c) 2022 Dell Inc, or its subsidiaries.
 
-all:
+all: buflint doc
+
+buflint:
+#	buf lint
```
[the remark this answers is not in the document]
Comment (sandersms) [reply network/opinetcommon/Makefile:8]: No - I will clean up the older comments that are residuals of working through the docker support.

File: network/opinetcommon/Makefile [hunk 2]
```diff
@@ -2,22 +2,26 @@
...
+	docker run --rm -v "${PWD}/../../common/v1":/common -v "${PWD}":/out -w /out bufbuild/buf lint
+
+bufgen:
+#	buf generate --template ../../buf.gen.yaml -o v1alpha1
+#	buf generate does not work in docker since it relies on locally installed protoc to compile the proto files
```
[the remark this answers is not in the document]
Comment (sandersms) [reply network/opinetcommon/Makefile:13]: Agreed now that the docker is working.

File: network/opinetcommon/Makefile [hunk 3]
```diff
@@ -2,22 +2,26 @@
...
+	docker run --rm -v "${PWD}/../../common/v1":/common -v "${PWD}/../..":/base -v "${PWD}":/out -w /out msandersdell/bufbuild-go-gen:1.1.0 generate --template /base/buf.gen.yaml -o v1alpha1
```
[the remark this answers is not in the document]
Comment (sandersms) [reply network/opinetcommon/Makefile:14]: Yes, it is a good idea to have them places somewhere to have control of any dep updates as the dockerfile that generates the image does specify the versions.  I didn't have a good location to place that currently and I will coordinate with Boris @glimchb  about where we want to place such items as well as a location in dockerhub to place the published image.  I am going to create a github repo for the dockerfile used to generate the image.

**Pull request merged**